include/zephyr/lorawan/lorawan.h

```diff
@@ -217,6 +217,15 @@ typedef uint8_t (*lorawan_battery_level_cb_t)(void);
  */
 typedef void (*lorawan_dr_changed_cb_t)(enum lorawan_datarate dr);
 
+/**
+ * @brief Defines the descriptor callback handler function signature.
+ *
+ * @param descriptor Descriptor field
+ *
+ * @return 0 if successful, negative errno code if failure
```

Comment:
The user is implementing this, so it should document what values should be returned in different circumstances.
What value do you return to proceed with the download, what value to terminate the download?